components/gcp/container/component_sdk/python/kfp_component/google/common/_utils.py

```diff
@@ -120,3 +123,55 @@ def wait_operation_done(get_operation, wait_interval):
             ))
         return operation
 
+
+def with_retries(
+    func: Callable,
+    on_error: Optional[Callable[[], Any]] = None,
+    errors: Tuple[Exception, ...] = Exception,
+    number_of_retries: int = 5,
+    delay: float = 1,
+):
+    """Retry decorator.
+
+    The decorator catches `errors`, calls `on_error` and retries after waiting `delay` seconds.
+
+    Args:
+        number_of_retries (int): Total number of retries if error is raised.
+        delay (float): Number of seconds to wait between consecutive retries.
+    """
+
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        remaining_retries = number_of_retries
+        while remaining_retries:
+            try:
+                return func(self, *args, **kwargs)
+            except errors as e:
+                remaining_retries -= 1
+                if not remaining_retries:
+                    raise
+
+                logging.warning(
+                    'Caught {}. Retrying in {} seconds...'.format(
+                        e._class__.__name__, delay
```

Comment:
@Ark-kun after trying out your changes I found this line to contain a bug. A second underscore needs to be prepended to __class__.

Reply:
Thank you. Fixed.